src/client/components/common/Combobox.tsx

```diff
@@ -0,0 +1,358 @@
+            backgroundColor="white"
+            borderRadius="5px"
+          >
+            <label {...getLabelProps()} hidden>
+              {label}
+            </label>
+            <InputGroup>
+              <Input
+                borderRadius={
+                  inputOptions?.useClearButton ? '5px 0px 0px 5px' : '5px'
+                }
+                focusBorderColor={
+                  searchResults.current.length > 0 ? '#3182ce' : 'error.500'
+                }
+                {...getInputProps({
+                  ...props,
+                  ref: inputRef,
+                  onFocus: () => {
+                    // reset field for new search
+                    setState({ inputValue: '' })
+                    searchResults.current = getTopSearchResults('')
+
+                    // ensure that menu should be displayed within window bounds
+                    dropdownDir = calculateDropdownDirection()
+
+                    // finally, toggle menu open
+                    openMenu()
+                    dropdownList.current?.scrollTo(0)
+                  },
+                })}
+              />
+              <InputRightElement pointerEvents="none">
+                <BiChevronDown aria-label="dropdown-icon" color="black" />
+              </InputRightElement>
+            </InputGroup>
+            {inputOptions?.useClearButton ? (
+              <IconButton
+                style={{ marginLeft: '-1px' }}
+                aria-label="Clear"
+                variant="outline"
+                borderRadius="0px 5px 5px 0px"
+                icon={<BiX size="16px" />}
+                onClick={() => {
+                  // runs input onFocus clear/reset functionality
+                  inputRef.current?.focus()
+                }}
+              />
+            ) : null}
+          </HStack>
+          <Box
+            boxShadow="0px 0px 10px rgba(216, 222, 235, 0.5)"
+            position={'absolute'}
+            maxH={`${dropdownHeight}px`}
+            top={dropdownDir === DropdownDirection.down ? '40px' : undefined}
+            bottom={dropdownDir === DropdownDirection.up ? '40px' : undefined}
+            bg="white"
+            w="100%"
+            zIndex={99}
+          >
+            <FixedSizeList
+              {...getMenuProps({
+                ref: dropdownList,
+              })}
+              height={
+                isOpen ? getDropdownHeight(searchResults.current.length) : 0
+              }
+              itemSize={itemHeight}
+              itemCount={searchResults.current.length}
+              itemData={
+                {
+                  items: searchResults.current,
+                  highlightedIndex: highlightedIndex,
+                  getItemProps: getItemProps,
+                  selectedItem: selectedItem,
+                  topDropdownInset: dropdownInset,
+                } as ItemRendererProps['data']
+              }
+              innerElementType={innerElementType}
```

Comment:
Do we need to set this prop? Isn't the default value a `div` and also the styling is applied on the `ListItem`?

Reply:
This prop is required to add the 8px in-list padding at the top and bottom as required in the design, as the row elements use absolute positioning within the div and thus require us to manually extend the height of the div element to account for the padding.